### PR TITLE
Update frontend to use and display real data from backend 

### PR DESCRIPTION
**Pull request by lin8231**
* All information local to a competition are now fetched in App.js whenever there is a logged in user and an existing comp id. This currently includes
    * Investments 
    * Trending
    * Recent buys 
    * General info (of a user in this competition, i.e. rank, networth...)
 * This info is fed as props to child components who require it 
 * Dashboard is now fully rendering backend data!
    * The styling has been updated slightly to reflect changes from placeholders to real data 
* Calculation of difference of sgonk from yesterday has been updated 
    * It now uses the last two datapoints in the datapoints array given from backend 
* Auth and userid flows are still hardcoded for the moment being - using the user with the most data to make coding easier

### CODE REVIEW

File: project/frontend/src/App.js [hunk 1]
```diff
@@ -37,11 +39,62 @@ function App() {
...
         ...prevState,
-        id: 123,
+        id: 1,
       };
     });
   }
 
+  React.useEffect(() => {
+    if (user.signedIn && user.id && compId) {
+      Promise.all([
```
Comment (leonnemets) [inline project/frontend/src/App.js:49]: what happens if one for them fails?

File: project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js [hunk 2]
```diff
@@ -2,11 +2,14 @@ import React from "react";
 import classes from "./ShortSGonksRow.module.css";
 
 const ShortSGonksRow = (props) => {
+  console.log(props);
   const getDiffIndicator = () => {
     const diff =
-      props.sgonk.currentPrice > props.sgonk.priceYesterday
+      props.datapoints[props.datapoints.length - 1] >
+      props.datapoints[props.datapoints.length - 2]
```
Comment (leonnemets) [inline project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js:9]: {nit} may be more readable if you do priceDelta = props.datapoints[props.datapoints.length - 1]  -  props.datapoints[props.datapoints.length - 2]
and then priceDelta < 0 ? ..  : priceDelta > 0 ? ... : ...

File: project/frontend/src/containers/Dashboard/Dashboard.js [hunk 3]
```diff
@@ -56,56 +57,43 @@ const Dashboard = (props) => {
...
-    "Fultondale al",
-    "Ben Askren",
-    "Stock-market",
-    "Etsy stock",
-    "Elliot Page",
-    "Susan Rice",
-    "Raya and the Last Dragon",
-    "Starlink",
-  ];
+  const toDayHourMinute = (totalTime) => {
```
Comment (leonnemets) [inline project/frontend/src/containers/Dashboard/Dashboard.js:60]: Is there a library do some of this date/time formatting?

File: project/frontend/src/containers/Dashboard/Dashboard.js [hunk 4]
```diff
@@ -56,56 +57,43 @@ const Dashboard = (props) => {
...
+    const millisNow = Date.now();
+    const remainingTime = props.generalInfo.endDate - millisNow;
+    return formatDHM(toDayHourMinute(remainingTime));
+  };
+
+  const getOrdinalOnly = (n) => {
+    const s = ["th", "st", "nd", "rd"],
+      v = n % 100;
+    return s[(v - 20) % 10] || s[v] || s[0];
+  };
```
Comment (leonnemets) [inline project/frontend/src/containers/Dashboard/Dashboard.js:96]: I have seen this one before, can it be factored out?

**Review by leonnemets — APPROVED** [review]
LGTM. Please address the comments

**Review by Phoebe-Khokgawe — APPROVED** [review]

File: project/frontend/src/components/Chart/Chart.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import classes from "./Chart.module.css";
+
+
+const Chart = (props) => {
```
[the remark this answers is not in the document]
Comment (lin8231) [reply project/frontend/src/components/Chart/Chart.js:5]: That was committed by Emma

File: project/frontend/src/App.js [hunk 6]
```diff
@@ -37,11 +39,62 @@ function App() {
...
+              return {
+                ...prevState,
+                recentBuys: data,
+              };
+            });
+          }),
+        fetch("./investments?user=" + user.id + "&competition=" + compId)
+          .then((response) => response.json())
+          .then((data) => {
+            setCompetitionInfo((prevState) => {
```
[the remark this answers is not in the document]
Comment (lin8231) [reply project/frontend/src/App.js:74]: So this is doesn't really have anything to do with components. All that block is doing is fetching all the data relevant to a specific competition from the backend (they're separate servlets) and saving it as state (the state called competitionInfo). These calls they're not overwriting the previous state, they're filling it with information with every call. Once everything is loaded the competitionInfo state should be an object like this:

competitionInfo: {
    generalInfo: {...},
    investments: {...},
    recentBuys: {...},
    trending: {...}
}

File: project/frontend/src/App.js (hunk 1 above)
Comment (lin8231) [reply project/frontend/src/App.js:49]: If some fail, then the ones that failed will not be displayed. The successful ones will still be fine.

File: project/frontend/src/containers/Dashboard/Dashboard.js (hunk 3 above)
Comment (lin8231) [reply project/frontend/src/containers/Dashboard/Dashboard.js:60]: There are no obvious ones that do what I want it to do, and I figured that less dependencies is better with something that's so easily done.

File: project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js (hunk 2 above)
Comment (lin8231) [reply project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js:9]: Can do

File: project/frontend/src/containers/Dashboard/Dashboard.js (hunk 4 above)
Comment (lin8231) [reply project/frontend/src/containers/Dashboard/Dashboard.js:96]: Hmm, I did think about that, I'm just not quite sure where to put it

File: project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js (hunk 2 above)
Comment (lin8231) [reply project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js:9]: Done

File: project/frontend/src/containers/Dashboard/Dashboard.js (hunk 4 above)
Comment (lin8231) [reply project/frontend/src/containers/Dashboard/Dashboard.js:96]: I guess I could make a Rank component?

Comment (lin8231) [reply project/frontend/src/containers/Dashboard/Dashboard.js]: Made a Rank component. In hindsight even though it's like a 2 line component, that was a really good idea.

**Review by texm — APPROVED** [review]

**Pull request merged**